File: src/code-samples/langchain/agents-steering.py
"""Agents docs: steering middleware example."""

# :snippet-start: agents-steering-py
from langchain.agents import create_agent
from langchain.agents.middleware import HumanInTheLoopMiddleware
from langchain.tools import tool


@tool
def search(query: str) -> str:
    """Search for a query and return a short summary."""
    return f"Search results for: {query}"


agent = create_agent(
    model="anthropic:claude-sonnet-4-6",
    tools=[search],
    middleware=[HumanInTheLoopMiddleware(interrupt_on={"write_file": True})],
)
# :snippet-end:


# :remove-start:
if __name__ == "__main__":
    assert agent is not None
    print("✓ agents steering sample compiles")
# :remove-end:

File: src/code-samples/langchain/agents-steering.ts
// Agents docs: steering middleware example.

// :snippet-start: agents-steering-js
import { createAgent, humanInTheLoopMiddleware, tool } from "langchain";
import * as z from "zod";

const search = tool(({ query }) => `Search results for: ${query}`, {
  name: "search",
  description: "Search for a query and return a short summary.",
  schema: z.object({ query: z.string() }),
});

const agent = createAgent({
  model: "anthropic:claude-sonnet-4-6",
  tools: [search],
  middleware: [humanInTheLoopMiddleware({ interruptOn: { writeFile: true } })],
});
// :snippet-end:

// :remove-start:
async function main() {
  void agent;
  console.log("✓ agents steering sample compiles");
}

main();
// :remove-end:

